services/121-service/src/activities/dtos/activities.dto.ts

```diff
@@ -1,92 +1,98 @@
+import { FinancialServiceProviderName } from '@121-service/src/financial-service-providers/enum/financial-service-provider-name.enum';
 import { MessageContentType } from '@121-service/src/notifications/enum/message-type.enum';
-import { NotificationType } from '@121-service/src/notifications/twilio.entity';
 import { TransactionStatusEnum } from '@121-service/src/payments/transactions/enums/transaction-status.enum';
 import { RegistrationStatusEnum } from '@121-service/src/registration/enum/registration-status.enum';
+import { LocalizedString } from '@121-service/src/shared/types/localized-string.type';
 
 export class ActivitiesDto {
   meta: {
     availableTypes: ActivityTypeEnum[];
     count: Partial<Record<ActivityTypeEnum, number>>;
   };
-  data: ActivityType[];
+  data: Activity[];
 }
 
-type ActivityType =
+export type Activity =
   | DataChangeActivity
+  | FinancialServiceProviderChangeActivity
   | MessageActivity
   | NoteActivity
   | StatusChangeActivity
   | TransactionActivity;
 
-enum ActivityTypeEnum {
+export enum ActivityTypeEnum {
   Transaction = 'transaction',
   Message = 'message',
   Note = 'note',
   DataChange = 'data-change',
   StatusChange = 'status-change',
-  FinancialServiceProviderChange = 'financial-service-provider-change', // ##TODO: Looks like this type is still missing in the rest of the code, e.g. extending from BaseActivity.
+  FinancialServiceProviderChange = 'financial-service-provider-change',
 }
 
 interface BaseActivity {
   id: string;
-  user: {
-    id: number;
-    username: string;
-  };
+  username?: string;
```

Comment:
Because of the way we've built the tables in the FE with primeng, the filters only work if they are on the top-level of an object (ie. not nested within an object within an object). This might become an issue in the future, but for the time being this was an easy fix in the BE, rather than having to do data manipulation in the frontend. But if I'm deviating from a convention, I'm happy to revert and add an extra step in the FE.

Reply:
Yes, this is deviating from a software architecture convention I am trying to get us to adhere to.
Unless that is not a good idea at all from API consumer perspective. If that is the case, then let's have a conversation. Otherwise, please revert and add the extra step. Assuming that is not very complex or a lot of work.